### ISSUE TITLE
[Bug]: TeamsAttachmentDownloader doesn't work in Bot Framework Emulator

### ISSUE THREAD

**Issue by qinezh** [microsoft/teams-ai#1154]
### Language

Javascript/Typescript

### Version

latest

### Description

I created a Teams bot app to handle inline image sent from users with teams-ai SDK. It works as expected in Teams client, however it doesn't work in Bot Framework Emulator and unhandled error is thrown: 

> *[onTurnError] unhandled error: TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be of type string or an instance of Buffer, ArrayBuffer, or Array or an Array-like Object. Received undefined*

Looks it's related to the implementation to check if content url starts with 'https://': https://github.com/microsoft/teams-ai/blob/8258ca0e2396c3f5bb376cdc62374b5cd11b5603/js/packages/teams-ai/src/TeamsAttachmentDownloader.ts#L76

In Bot Framework Emulator, the image is hosted from localhost with http endpoint:
![image](https://github.com/microsoft/teams-ai/assets/15644078/b6dd7cba-a0fe-4fc3-ad59-798ab5325acc)

In addition, I found `TeamsAttachmentDownloader` will try to get access token before downloading the attachment. When I test the bot app in Bot Framework Emulator with empty string of BOT_ID and BOT_PASSWORD, the step to get access token would be failed:

https://github.com/microsoft/teams-ai/blob/8258ca0e2396c3f5bb376cdc62374b5cd11b5603/js/packages/teams-ai/src/TeamsAttachmentDownloader.ts#L113-L128

We may consider skipping to get access token if the BOT_ID is empty, similar to the implementation of botbuilder SDK.

### Reproduction Steps

```bash
1. Create a Teams bot app with below code snippet.


const downloader = new TeamsAttachmentDownloader({
  botAppId: process.env.BOT_ID,
  botAppPassword: process.env.BOT_PASSWORD,
});

app.activity(ActivityTypes.Message, async (context: TurnContext, state: ApplicationTurnState) => {
    const attachments = await downloader.downloadFiles(context, state);
    await context.sendActivity(`you have uploaded ${attachments?.length ?? 0} attachments`);
});


2. Run the Teams bot app locally.
3. Test the bot app by uploading an image in Bot Framework Emulator.
```

**Comment by corinagum** [comment]
Hi @qinezh, from what you wrote it looks like you wrote some custom functionality around images and `TeamsAttachmentDownloader`. However, Bot Framework Emulator is not intended to test channel-unique features - it was originally created / designed to test features that are strictly adhering to the Bot Framework Protocol, and not unique to any one channel. 

Bot Framework is no longer in active development, and because of this and the above, there isn't any plan to extend Emulator to work with Teams features. 

The Teams AI team strongly recommends using Teams Toolkit to test your features, since any Teams feature will require the Teams client to test. You can take any of our samples and add the Teams Toolkit (TTK) `.yml` and (if you plan to deploy) the infra/azure files and place them in your project. You will also want to copy the `.env` folder into your project. Once you install the TTK VS Code extension, TTK will help you run the sample and automatically sideload the app into the Teams client. It's extremely convenient and I highly recommend!

**Comment by qinezh** [comment]
@corinagum thanks for your quick response and recommendation for Teams Toolkit. And I'm from Teams Toolkit dev team😄 

Recently, we have released a feature in Teams Toolkit called Teams App Test Tool: [Introducing Teams App Test Tool](https://devblogs.microsoft.com/microsoft365dev/introducing-teams-app-test-tool/). It’s a feature integrated into Teams Toolkit that helps developers debug Teams Bot application in an emulated environment similar to BF Emulator but optimized for Teams use cases. 

While investigating the support of multimodal scenario with teams-ai in Teams App Test Tool, we found this issue and the root cause is the same as the one encountered in the BF emulator. The issue was initially raised using the example of BF emulator, however the actual goal to support debugging bot app with teams-ai in Teams App Test Tool.

Can you please take another look about this issue?

**Comment by corinagum** [comment]
@qinezh haha my apologies, and thanks for the extra context; it helps a lot. That's exciting work I'm sure many people will love.

I've reopened the issue and will get back to you asap :)

**Comment by corinagum** [comment]
Hi @qinezh, thanks for your patience. I investigated this bug and I've confirmed the requests you are making regarding running a local bot. To reiterate, there are 2 direct problems:

1.  Check url for https or http://localhost
2. Don't perform an access token request if there's no app id and password.

I am fairly confident that both of these changes can be made without problems, but before starting I want to check in with my team first.

I also discovered another bug, linked above. I will start working on these issues once I've confirmed the work items with my team.
